code/game/gamemodes/gang/gang.dm

```diff
@@ -247,6 +248,20 @@ var/list/gang_colors_pool = list("red","orange","yellow","green","blue","purple"
 		gang_bosses += G.bosses
 	return gang_bosses
 
+/datum/game_mode/proc/shuttle_check()
+	if(forced_shuttle)
+		return
+	var/normies = 0 
+	for(var/mob/living/M in GLOB.player_list)
+		if(isliving(M) && !M.stat==DEAD && (!M.mind in get_all_gangsters())
```

Comment:
@Robustin the point is you're running ```get_all_gangsters()``` for every single player in the player list (which is all of them)

There's literally no reason for this, just where you have ```var/normies = 0``` outside the loop, you can define: ```var/list/cool_dudes = get_all_gangsters()``` and then you can change this to: ```if( ... && (!M.mind in cool_dudes))```

No functionality changes, Much more efficient code.